NumLib/TimeStepping/Algorithms/EvolutionaryPIDcontroller.cpp

```diff
@@ -104,6 +109,28 @@ bool EvolutionaryPIDcontroller::next(const double solution_error)
     return true;
 }
 
+double EvolutionaryPIDcontroller::limitStepSize(
+    const double h_new, const double h_n,
+    const bool previous_step_accepted) const
+{
+    const double h_in_range = std::max(_h_min, std::min(h_new, _h_max));
+    double limited_h =
+        std::max(_rel_h_min * h_n, std::min(h_in_range, _rel_h_max * h_n));
```

Comment:
Added a documentation to this member, and also added a details description for each constraint.